sale_contact_type/views/sale_order.xml

```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record id="view_order_form" model="ir.ui.view">
+        <field name="name">sale.order.form</field>
+        <field name="model">sale.order</field>
+        <field name="inherit_id" ref="sale.view_order_form"/>
+        <field name="arch" type="xml">
+            <xpath expr='//field[@name="partner_id"]' position='attributes'>
+                <attribute name="domain">[('customer', '=', True), '|', ('type', '=', 'ordering'), '&amp;', ('parent_id', '=', False), '|', ('child_ids', '=', False), ('has_ordering_contact_child', '!=', True)]</attribute>
```

Comment:
Maybe it can be reduced to:
```suggestion
                <attribute name="domain">[('customer', '=', True), '|', ('type', '=', 'ordering'), '&amp;', ('parent_id', '=', False), ('has_ordering_contact_child', '!=', True)]</attribute>
```

Reply:
Yes, I think so